src/h2olog/h2olog.cc

```diff
@@ -125,6 +126,44 @@ static void show_process(pid_t pid)
     infof("Attaching pid=%d (%s)", pid, cmdline);
 }
 
+static void drop_root_privilege(void)
+{
+    if (getuid() == 0) {
+        if (getgid() == 0) {
+            const char *sudo_gid = getenv("SUDO_GID");
+            if (sudo_gid == NULL) {
+                fprintf(stderr, "Error: failed to read the SUDO_GID env variable\n");
+                exit(EXIT_FAILURE);
+            }
+            errno = 0;
+            gid_t gid = (gid_t)strtol(sudo_gid, NULL, 10);
+            if (errno != 0) {
+                fprintf(stderr, "Error: overflow while parsing SUDO_GID\n");
```

Comment:
```suggestion
                fprintf(stderr, "Error: failed to parse SUDO_GID\n");
```
... or something alike. I think that the parsing might fail for reasons other than overflow?